reproject/common.py

```diff
@@ -116,6 +117,18 @@ def _reproject_dispatcher(
         If this is set to 'pil_image', a PIL ``Image`` object is returned. The
         'pil_image' option can only be used if the input was RGB images or if
         the input data has shape (3, ny, nx) and contains values between 0 and 255.
+    dask_method : {'memmap', 'none'}
+        Method to use when input array is a dask array. The methods are:
+            * ``'memmap'``: write out the entire input dask array to a temporary
+              memory-mapped array. This requires enough disk space to store
+              the entire input array, but should avoid accidentally loading
+              the entire array into memory.
+            * ``'none'``: load the dask array into memory as needed. This may
+              result in the entire array being loaded into memory. However,
+              this can be efficient under two conditions: if the array easily
+              fits into memory (as this will then be faster than ``'memmap'``),
+              and when the data contains more dimensions than the input WCS and
+              the block_size is chosen to iterate over the extra dimensions.
```

Comment:
(Not in this PR necessarily) Would it possible to detect the latter case and do this automatically?